src/core/core-varlink.c

```diff
@@ -555,6 +559,20 @@ static int vl_method_get_memberships(sd_varlink *link, sd_json_variant *paramete
         return sd_varlink_error(link, "io.systemd.UserDatabase.NoRecordFound", NULL);
 }
 
+static int vl_method_describe(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        Manager *manager = ASSERT_PTR(userdata);
+        int r;
+
+        assert(parameters);
+
+        r = manager_build_json(manager, &v);
+        if (r < 0)
+                return log_error_errno(r, "Failed to build manager JSON data: %m");
+
+        return sd_varlink_replybo(link, SD_JSON_BUILD_PAIR_VARIANT("manager", v));
```

Comment:
hmm, what made you insert "manager" as intermediary object here, rather than directly return this? in hostnamed's Describe call we don't do that, so why here?

Reply:
I was following the BootControl interface which does the same.